pkg/manifests/bundleloader.go

```diff
@@ -0,0 +1,177 @@
+	return func(path string, f os.FileInfo, err error) (walkErr error) {
+		if walkErr = walk(path, f, err); walkErr != nil && walkErr != filepath.SkipDir {
+			*errs = append(*errs, walkErr)
+			return nil
+		}
+
+		return walkErr
+	}
+}
+
+func (b *bundleLoader) LoadBundleWalkFunc(path string, f os.FileInfo, err error) error {
+	if f == nil {
+		return fmt.Errorf("invalid file: %v", f)
+	}
+
+	if f.IsDir() {
+		if strings.HasPrefix(f.Name(), ".") {
+			return filepath.SkipDir
+		}
+		return nil
+	}
+
+	if strings.HasPrefix(f.Name(), ".") {
+		return nil
+	}
+
+	fileReader, err := os.Open(path)
+	if err != nil {
+		return fmt.Errorf("unable to load file %s: %s", path, err)
+	}
+
+	decoder := yaml.NewYAMLOrJSONDecoder(fileReader, 30)
+	csv := unstructured.Unstructured{}
+
+	if err = decoder.Decode(&csv); err != nil {
+		return nil
+	}
+
+	if csv.GetKind() != operatorsv1alpha1.ClusterServiceVersionKind {
+		return nil
+	}
+
+	var errs []error
+	bundle, err := loadBundle(csv.GetName(), filepath.Dir(path))
+	if err != nil {
+		errs = append(errs, fmt.Errorf("error loading objs in directory: %s", err))
+	}
+
+	if bundle == nil || bundle.CSV == nil {
+		errs = append(errs, fmt.Errorf("no bundle csv found"))
+		return utilerrors.NewAggregate(errs)
+	}
+
+	b.bundle = bundle
+
+	return utilerrors.NewAggregate(errs)
+}
+
+// loadBundle takes the directory that a CSV is in and assumes the rest of the objects in that directory
+// are part of the bundle.
+func loadBundle(csvName string, dir string) (*Bundle, error) {
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	var errs []error
+	bundle := &Bundle{
+		Name: csvName,
+	}
+	for _, f := range files {
+		if f.IsDir() {
+			continue
+		}
+
+		if strings.HasPrefix(f.Name(), ".") {
+			continue
+		}
+
+		path := filepath.Join(dir, f.Name())
+		fileReader, err := os.Open(path)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("unable to load file %s: %s", path, err))
+			continue
+		}
+
+		decoder := yaml.NewYAMLOrJSONDecoder(fileReader, 30)
+		obj := &unstructured.Unstructured{}
+		if err = decoder.Decode(obj); err != nil {
+			continue
+		}
+
+		bundle.Objects = append(bundle.Objects, obj)
+
+		switch kind := obj.GetKind(); kind {
+		case "ClusterServiceVersion":
+			if bundle.CSV != nil {
+				return nil, fmt.Errorf("invalid bundle: contains multiple CSVs")
+			}
+			csv := operatorsv1alpha1.ClusterServiceVersion{}
+			err := runtime.DefaultUnstructuredConverter.FromUnstructured(
+				obj.UnstructuredContent(),
+				&csv)
+			if err != nil {
+				return nil, fmt.Errorf("unable to parse CSV %s: %s", f.Name(), err.Error())
+			}
+			bundle.CSV = &csv
+		case "CustomResourceDefinition":
+			version := obj.GetAPIVersion()
+			if version == apiextensionsv1beta1.SchemeGroupVersion.String() {
+				crd := apiextensionsv1beta1.CustomResourceDefinition{}
+				err := runtime.DefaultUnstructuredConverter.FromUnstructured(
```

Comment:
This string-to-unstructured-to-crd conversion is the root cause of `int64 to float64` error. So we should avoid converting `unstructured` to a specific type. Should only use `unstructured` to determine the object type and then convert string data straight to the object type like CRD. See: https://github.com/operator-framework/operator-registry/pull/334/files#diff-b754381164690e8ba263f719136845d3R320

Reply:
Is this behavior defined for unstructured conversion?

Reply:
I think this issue can be addressed in a follow-up.

Reply:
I'm fine with fixing this at another time. Please make a note somewhere so it won't get lost.